src/main/java/com/minecolonies/coremod/entity/ai/citizen/netherworker/EntityAIWorkNether.java

```diff
@@ -0,0 +1,958 @@
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Put together the valid list of things to request for food
+     */
+    private List<ItemStack> getEdiblesList()
+    {
+        final List<ItemStorage> allowedItems = getOwnBuilding().getModuleMatching(ItemListModule.class, m -> m.getId().equals(FOOD_EXCLUSION_LIST)).getList();
+        netherEdible.removeIf(item -> allowedItems.contains(new ItemStorage(item)));
+        return netherEdible;
+    }
+
+    /**
+     * Check if we need to request food, and do so if necessary.
+     */
+    protected void checkAndRequestFood(int itemCount)
+    {
+        List<ItemStack> edibleList = getEdiblesList();
+        final IDeliverable edible = new StackList(edibleList, "Edible Food", itemCount);
+
+        List<ItemStack> foodInInv = InventoryUtils.getContainedFromItemHandler(edibleList, worker.getItemHandlerCitizen());
+
+        double availSaturation = 0;
+        for(ItemStack item : foodInInv)
+        {
+            //int amount = (int) Math.round(Math.ceil((requiredSaturation - foundSaturation) / (float) itemFood.getNutrition()));
+
+            // Figure out what the total saturation value of the stack is
+            final Food itemFood = item.getItem().getFoodProperties();
+            final double satIncrease = itemFood.getNutrition() * (1.0 + worker.getCitizenColonyHandler().getColony().getResearchManager().getResearchEffects().getEffectStrength(SATURATION)) * item.getCount();
+
+            // update the running total
+            availSaturation +=  satIncrease;
+        }
+
+        if(availSaturation >= AVERAGE_SATURATION)
+        {
+            return;
+        }
+
+        if(InventoryUtils.getItemCountInProvider(getOwnBuilding(), item -> edible.matches(item)) < itemCount)
+        {
+            if (edible != null
+                    && !getOwnBuilding().hasWorkerOpenRequestsOfType(-1, TypeToken.of(edible.getClass()))
+                    && !getOwnBuilding().hasWorkerOpenRequestsOfType(worker.getCitizenData().getId(), TypeToken.of(edible.getClass())))
+            {
+                Log.getLogger().info("Requesting food, with available saturation: " + availSaturation);
+                getOwnBuilding().createRequest(edible, true);
+            }
+        }
+        else
+        {
+            int amountToTransfer = (int) (AVERAGE_SATURATION - availSaturation);
+            Log.getLogger().info("Picking up food for " + amountToTransfer + " saturation");
+            InventoryUtils.transferFoodUpToSaturation((IBuilding) getOwnBuilding(), worker.getInventoryCitizen(), amountToTransfer, stack -> edible.matches(stack));
+        }
+    }
+
+    /**
+     * Attempt to eat to restore some saturation
+     */
+    protected void attemptToEat()
+    {
+        final IDeliverable edible = new StackList(getEdiblesList(), "Edible Food", 1);
```

Comment:
A few of the methods down here seem like we got very similar utilities already (i.e. in the eating AI) we should potentially opt into attempting to reuse those

Reply:
Some of them were lifted from there with minor modifications. I tried to reuse them, but that attempt made quite a mess out of the eating AI, because of the "during simulation" vs "during statemachine" differences.